File: ruby/bob/bob.rb
class Bob
	def self.hey(string)
		string = string.gsub(/\s+/, "")
		if string == ''
			'Fine. Be that way!'
		elsif string.upcase == string && string[-1] == '?' && string.count("a-zA-Z") != 0
			"Calm down, I know what I'm doing!"
		elsif string.upcase == string && string.count("a-zA-Z") != 0
			'Whoa, chill out!'
		elsif string[-1] == '?'
			'Sure.'
		else
			'Whatever.'
		end
	end
end
module BookKeeping
	VERSION = 2
end

File: javascript/bob/bob.js
var Bob = function() {
  this.hey = function(string) {
		string = string.trim();
		if(string == ''){
			return 'Fine. Be that way!';
		} else if(string.slice(-1) == '?' && (string.toUpperCase() != string)){
			return 'Sure.';
		} else if(string.slice(-1) == '?' && /[^a-z]*$/.test(string)){
			return "Calm down, I know what I'm doing!";
		}else if(string.toUpperCase() == string){
			return 'Whoa, chill out!';
		} else{
			return 'Whatever.';
		}
  };
};
module.exports = Bob;